cmd/paasta-metastatus/main.go

```diff
@@ -0,0 +1,224 @@
+	flag.CommandLine.SortFlags = false
+	flag.Parse()
+	return nil
+}
+
+func printDashboards(
+	cluster string, dashboards map[string]interface{}, sb *strings.Builder,
+) {
+	if dashboards == nil {
+		sb.WriteString(aurora.Red("No dashboards configured!\n").String())
+	} else {
+		spacing := 0
+		for svc := range dashboards {
+			svcLen := len(svc)
+			if svcLen > spacing {
+				spacing = svcLen
+			}
+		}
+		spacing++
+		for svc, dashboard := range dashboards {
+			spacer := strings.Repeat(" ", spacing-len(svc))
+			sb.WriteString(fmt.Sprintf("  %s:%s", svc, spacer))
+			switch d := dashboard.(type) {
+			case string:
+				sb.WriteString(aurora.Cyan(d).String())
+			case []interface{}:
+				if len(d) > 1 {
+					for _, url := range d {
+						sb.WriteString(fmt.Sprintf("\n    %v", aurora.Cyan(url.(string))))
+					}
+				} else {
+					sb.WriteString(aurora.Cyan(d[0].(string)).String())
+				}
+			}
+			sb.WriteString("\n")
+		}
+	}
+}
+
+func getMetastatusCmdArgs(opts *PaastaMetastatusOptions) ([]string, time.Duration) {
+	cmdArgs := []string{}
+	verbosity := 0
+	timeout := time.Duration(20)
+
+	if opts.Verbosity > 0 {
+		verbosity = opts.Verbosity
+		timeout = 120
+	}
+	if opts.AutoscalingInfo {
+		cmdArgs = append(cmdArgs, "-a")
+		if verbosity < 2 {
+			verbosity = 2
+		}
+		timeout = 120
+	}
+	if len(opts.Groupings) > 0 {
+		cmdArgs = append(cmdArgs, "-g")
+		cmdArgs = append(cmdArgs, opts.Groupings...)
+	}
+	if opts.UseMesosCache {
+		cmdArgs = append(cmdArgs, "--use-mesos-cache")
+	}
+	if verbosity > 0 {
+		cmdArgs = append(cmdArgs, fmt.Sprintf("-%s", strings.Repeat("v", verbosity)))
+	}
+
+	return cmdArgs, timeout
+}
+
+func printAPIStatus(cluster, endpoint string, opts *PaastaMetastatusOptions, sb *strings.Builder) error {
+	url, err := url.Parse(endpoint)
+	if err != nil {
+		return fmt.Errorf("Failed to parse API endpoint %v: %v", endpoint, err)
+	}
+
+	var (
+		transport        = httptransport.New(url.Host, apiclient.DefaultBasePath, []string{url.Scheme})
+		client           = apiclient.New(transport, strfmt.Default)
+		cmdArgs, timeout = getMetastatusCmdArgs(opts)
+	)
+
+	ctx, cancel := context.WithTimeout(context.Background(), timeout*time.Second)
+	defer cancel()
+	mp := &operations.MetastatusParams{CmdArgs: cmdArgs, Context: ctx}
+	resp, err := client.Operations.Metastatus(mp)
+	if err != nil {
+		return fmt.Errorf("Failed to get metastatus: %s", aurora.Red(err))
+	}
+	sb.WriteString(fmt.Sprintf("%s\n", resp.Payload.Output))
+	return nil
+}
+
+func printClusterStatus(
+	cluster, endpoint string,
+	dashboards map[string]interface{},
+	opts *PaastaMetastatusOptions,
+) bool {
+	sb := &strings.Builder{}
+	sb.WriteString(fmt.Sprintf("Cluster: %v\n", cluster))
+	printDashboards(cluster, dashboards, sb)
+	success := true
+	err := printAPIStatus(cluster, endpoint, opts, sb)
+	if err != nil {
+		success = false
+		sb.WriteString(fmt.Sprintf("Failed to get status for cluster %v: %v\n", cluster, err))
+	}
+	fmt.Print(sb.String())
```

Comment:
I'm kind of reluctant to dump to stdout from here. what do you think of giving this thing an io.Writer as a param and writing to that and reading back out in `metastatus`?

Reply:
or we could just return `sb`?